packages/core/src/delegation/DelegationNode.ts

```diff
@@ -129,6 +129,41 @@ export default class DelegationNode extends DelegationBaseNode
     return query(this.parentId)
   }
 
+  public async findParent(
+    address: string
+  ): Promise<{
+    steps: number
+    node: DelegationNode | null
+  }> {
+    if (typeof this.parentId === 'undefined') {
+      return {
+        steps: 0,
+        node: null,
+      }
+    }
+
+    let delegationTreeTraversalSteps = 0
+    let delegationNode = await DelegationNode.query(this.parentId)
+
+    // while there is a parent and we have a node but not the right node ...
+    while (
+      delegationNode !== null &&
+      typeof delegationNode.parentId !== 'undefined' &&
+      delegationNode?.account !== address
+    ) {
+      delegationTreeTraversalSteps += 1
+
+      // this 'await in loop' is unavoidable because we need the node for the next await
```

Comment:
I guess recursively attaching `.then()` clauses is the only other option and that may hit the call stack limit at some point

Reply:
(recursion instead of a `while`)

Reply:
I wouldn't make the code worse only to avoid a lint error. 😅 I couldn't come up with a solution that wouldn't make the code worse.